include/jsonv/serialization_boost.hpp

```diff
@@ -0,0 +1,54 @@
+/** \file jsonv/serialization_boost.hpp
+*  Specialization types for serialization of boost classes and containers.
+*  These are usually not needed unless you are writing your own
+*  \c extractor or \c serializer.
+*
+*  Copyright (c) 2015 by Travis Gockel. All rights reserved.
+*
+*  This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+*  as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+*  version.
+*
+*  \author Vladimir Venediktov (vvenedict@gmail.com)
+**/
+
+#ifndef __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
+#define __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
+
+#include <jsonv/serialization_util.hpp>
+#include <boost/optional.hpp>
```

Comment:
Instead of using `boost::optional<T>` directly, I would prefer doing something like I did with `string_view` -- allow choosing the `optional` implementation. There's a small block of macro in `string_view.hpp`:

```
#ifndef JSONV_STRING_VIEW_TYPE
#   if defined(JSONV_STRING_VIEW_USE_STD) && JSONV_STRING_VIEW_USE_STD
#       define JSONV_STRING_VIEW_TYPE    std::string_view
#       define JSONV_STRING_VIEW_INCLUDE <string_view>
#   elif defined(JSONV_STRING_VIEW_USE_BOOST) && JSONV_STRING_VIEW_USE_BOOST
#       define JSONV_STRING_VIEW_TYPE    boost::string_ref
#       define JSONV_STRING_VIEW_INCLUDE <boost/utility/string_ref.hpp>
#   else
#       define JSONV_STRING_VIEW_TYPE    jsonv::detail::string_view
#       define JSONV_STRING_VIEW_INCLUDE <jsonv/detail/string_view.hpp>
#   endif
#endif
```

Do the same for `optional<T>` (although don't bother writing a `jsonv::detail::optional<T>`).